django/core/validators.py

```diff
@@ -113,11 +115,24 @@ def __init__(self, schemes=None, **kwargs):
 
     def __call__(self, value):
         value = force_text(value)
-        # Check first if the scheme is valid
         scheme = value.split('://')[0].lower()
-        if scheme not in self.schemes:
+
+        # A naive check for valid schemes
+        # schemes may contain an optional +, but only if followed
+        # by one or more characters
+        # RFC 3986 has the complete details
+        # http://tools.ietf.org/html/rfc3986#section-3.1
+        scheme_regex = _lazy_re_compile(r'^[a-z][a-z0-9]+(\+[a-z0-9]+)?$')
```

Comment:
I'm a bit confused here; so either we do some basic checks no matter what schemes we are validating - this would mean moving the regular expression check higher in the queue; but then if the magic value `__all__` is passed, we accept all schemes even if they are not matching the regular expression.

Did I understand it correctly? I thought previously it was that even if `__all__` is passed, we should still do some sanity checks to warn about any "obvious" incorrect schemes.